src/main.rs

```diff
@@ -22,6 +24,43 @@ mod dvi;
 mod framebuffer;
 mod link;
 
+global_asm! {
+    "
+    .section .text
+    .align 4
+    data_cpy_table:
+     .word _scratch_x_source
+     .word _scratch_x_start
+     .word _scratch_x_end
+     .word _scratch_y_source
+     .word _scratch_y_start
+     .word _scratch_y_end
+     .word 0
+    .global __pre_init
+    .type __pre_init,%function
+    .thumb_func
+    __pre_init:
```

Comment:
Is this symbol expected somewhere? how is this called?

Reply:
It is called from cortex-m-rt on startup, before transferring control to main. That is documented in [objdump symbols](https://docs.rs/cortex-m-rt/latest/cortex_m_rt/index.html#symbols-objdump-nm), and there is more discussion in [[#pre_init]](https://docs.rs/cortex-m-rt/latest/cortex_m_rt/attr.pre_init.html). Writing it in asm is a way to guarantee that no undefined behavior will be invoked.